analyzers/src/SonarAnalyzer.CSharp/Trackers/CSharpAssignmentFinder.cs

```diff
@@ -30,13 +30,15 @@ public class CSharpAssignmentFinder : AssignmentFinder
         protected override SyntaxNode GetTopMostContainingMethod(SyntaxNode node) =>
             node.GetTopMostContainingMethod();
 
-        protected override bool IsAssignmentToIdentifier(SyntaxNode node, string identifierName, out SyntaxNode rightExpression)
+        protected override bool IsAssignmentToIdentifier(SyntaxNode node, string identifierName, bool anyAssignmentKind, out SyntaxNode rightExpression)
         {
             if (node is ExpressionStatementSyntax statement)
             {
                 node = statement.Expression;
             }
-            if (node is AssignmentExpressionSyntax assignment && assignment.Left.NameIs(identifierName))
+            if ((anyAssignmentKind || node.IsKind(SyntaxKind.SimpleAssignmentExpression))
```

Comment:
could you add a comment above to say what this logic is for? (maybe give an example of another SyntaxKind which is an AssignmentExpressionSyntax but not simple assignment